### PR TITLE
Document Trusted Root Certificate support w/VMware Event Router

### LINKED ISSUE #383: Document procedure for VEBA to add a trusted root CA
**Describe the bug**
Document and verify steps to support #370

### PR DESCRIPTION
**Pull request by lamw**
Signed-off-by: William Lam <wlam@vmware.com>

## Pull Request Checklist
🚨 Please review the [guidelines for contributing](https://vmweventbroker.io/community) to this repository.

- [ ] Please ensure that you are making a pull request against the **Development** branch
- [ ] Please use the `WIP` keyword in the title of your PR if you are not ready for review
- [ ] Please ensure that you have opened a Github Issue if you are resolving/fixing a problem
- [ ] Please ensure that you have [signed](https://help.github.com/en/github/authenticating-to-github/signing-commits) all commits and that you have [squashed](https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec) all relevant commits related to your change
- [ ] Please make sure that you have tested your change locally by successfully [building and deploying the VMware Event Broker Appliance](https://vmweventbroker.io/kb/contribute-appliance) and/or [building and deploying VMware Event Router](https://vmweventbroker.io/kb/contribute-eventrouter)
- [ ] Please include any relevant screenshots and/or output as part of your testing
- [ ] Please include any documentation updates that is applicable for your changes

## Change Type

What types of changes does your code introduce to the VMware Event Broker Appliance?

_Put an `x` in all boxes that apply_

Please check the type of change your PR introduces:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation changes
- [ ] Other (please describe):

## Resolved Issues

Closes #383 

## Testing Verification

* Successfully built VEBA appliance w/updated VMware Event Router and successfully reconfigure to add trusted root certificate

## Additional Information

N/A

### CODE REVIEW

File: docs/kb/advanced-certificates.md [hunk 1]
```diff
@@ -63,3 +63,98 @@ LS0tLS1CRUd......==
...
+```console
+kubectl -n vmware-system get pods
+NAME                                   READY   STATUS    RESTARTS   AGE
+tinywww-dd88dc7db-q6pzw                1/1     Running   0          166m
+veba-rabbit-server-0                   1/1     Running   0          167m
+veba-ui-8d6584f84-ckvwb                1/1     Running   0          166m
+vmware-event-router-7759d8bffc-45c92   1/1     Running   0          37m
+```
+
+Lastly, you can also look at the VMware Event Router log and ensure you are no should seeing the message `potentially insecure connection to vCenter`, which is what you would see if the connection was not trusted if you had TLS verified disabled.
```
Comment (embano1) [inline docs/kb/advanced-certificates.md:153]: This is a bit confusing as the output you show is actually saying it's insecure. You might want to move that paragraph above to the intro section to show that this example/doc fixes this behavior. At the end of this section, you can show the output of the router again without this line being printed.

File: docs/kb/advanced-certificates.md [hunk 2]
```diff
@@ -63,3 +63,98 @@ LS0tLS1CRUd......==
...
+* Pre-download you root certificates from your internal certificate, which must be .PEM encoded
+
+In the example, the downloaded root certificate file is named `ca.crt`
+
+### Steps
+
+Step 1 - Transfer or copy the contents of the root certificate to the VMware Event Broker Appliance and create a new kubernetes configMap from that file.
+
+```console
+kubectl -n vmware-system create cm ca-cert --from-file ca.crt
```
Comment (embano1) [inline docs/kb/advanced-certificates.md:86]: Below the file name is vc-ca.cert which should not have worked in your case then...

File: docs/kb/advanced-certificates.md [hunk 3]
```diff
@@ -63,3 +63,98 @@ LS0tLS1CRUd......==
...
+...
+metricsProvider:
+  default:
+    bindAddress: 0.0.0.0:8082
+  name: veba-metrics
+  type: default
+certificates:
+  rootCAs:
+    - /etc/ssl/certs/ca-certificates.crt
+    - /etc/vmware-event-router/ssl/vc-ca.crt
```
Comment (embano1) [inline docs/kb/advanced-certificates.md:130]: See comment above

File: docs/kb/advanced-certificates.md [hunk 4]
```diff
@@ -63,3 +63,98 @@ LS0tLS1CRUd......==
...
+```
+
+Step 2 - Undeploy the VMware Event Router and delete the current configuration secret.
+
+```console
+kubectl -n vmware-system delete -f /root/config/event-router-k8s.yaml
+kubectl -n vmware-system delete secret event-router-config
+```
+
+Step 3 - Edit `/root/config/event-router-k8s.yaml` and append the additional volumenMount and volume entry as shown in the snippet below.
```
Comment (embano1) [inline docs/kb/advanced-certificates.md:96]: Might want to syntax highlight volume and volumeMount. Also typo in volumenMount.

File: docs/kb/advanced-certificates.md (hunk 1 above)
Comment (embano1) [inline docs/kb/advanced-certificates.md:153]: typo: You are not should ...

File: docs/kb/advanced-certificates.md [hunk 5]
```diff
@@ -63,3 +63,98 @@ LS0tLS1CRUd......==
...
+
+For organizations that require the use of a Private or Enterprise Certificate Authority (CA) and wish to have full TLS trust when the VMware Event Broker Appliance is configured with a vCenter Server that has a TLS certificate generated by the internal CA, a root of trust must be established.
+
+To support this use case, a post-deployment reconfiguration of the VMware Event Broker Appliance is required to import the trusted root certificate of your internal certificate authority. 
+
+In order to import your trusted root certificate, please follow the steps outlined below.
+
+### Assumptions
+
+* Pre-download you root certificates from your internal certificate, which must be .PEM encoded
```
Comment (embano1) [inline docs/kb/advanced-certificates.md:77]: you -> your

**Review by embano1 — CHANGES_REQUESTED** [review]

File: docs/kb/advanced-certificates.md (hunk 1 above)
Comment (lamw) [reply docs/kb/advanced-certificates.md:153]: Make sense.

File: docs/kb/advanced-certificates.md (hunk 2 above)
Comment (lamw) [reply docs/kb/advanced-certificates.md:86]: This was a copy/paste error for docs. Although we used vCenter's root CA, customers would be using their actual root CA, so I didn't want to call this `vc-ca` but rather `ca-cert`

File: docs/kb/advanced-certificates.md [hunk 6]
```diff
@@ -63,3 +63,103 @@ LS0tLS1CRUd......==
...
+...
+metricsProvider:
+  default:
+    bindAddress: 0.0.0.0:8082
+  name: veba-metrics
+  type: default
+certificates:
+  rootCAs:
+    - /etc/ssl/certs/ca-certificates.crt
+    - /etc/vmware-event-router/ssl/ca-cert.crt
```
Comment (embano1) [inline docs/kb/advanced-certificates.md:135]: IMHO this is still wrong. If the filename above is `ca.crt` the projected file name would be the same, no?

Comment (lamw) [reply docs/kb/advanced-certificates.md]: I think I fully grok it now, I may have miss-understood on the naming. I've made it a bit more clear now. Let me know if this is good

**Review by embano1 — APPROVED** [review]
Wonderful!

**Pull request merged**